packages/roosterjs-editor-core/lib/coreAPI/selectRange.ts

```diff
@@ -1,73 +1,55 @@
-                        currentRange.startOffset == range.startOffset &&
-                        currentRange.endContainer == range.endContainer &&
-                        currentRange.endOffset == range.endOffset
-                    ) {
-                        needAddRange = false;
-                    } else {
-                        selection.removeAllRanges();
-                    }
-                } catch (e) {}
-            }
-
-            if (needAddRange) {
-                selection.addRange(range);
-            }
-
-            if (!hasFocus(core)) {
-                core.cachedSelectionRange = range;
-            }
-
-            return true;
-        }
-    }
-
-    return false;
-};
-
-export default select;
+import EditorCore, { Select, SelectRange } from '../interfaces/EditorCore';
+import hasFocus from './hasFocus';
+import { Browser, contains, createRange } from 'roosterjs-editor-dom';
+
+const selectRange: SelectRange = (core: EditorCore, range: Range, skipSameRange?: boolean) => {
+    if (contains(core.contentDiv, range)) {
+        let selection = core.document.defaultView.getSelection();
+        if (selection) {
+            let needAddRange = true;
+
+            if (selection.rangeCount > 0) {
+                // Workaround IE exception 800a025e
+                try {
+                    // Do not remove/add range if current selection is the same with target range
+                    // Without this check, execCommand() may fail in Edge since we changed the selection
+                    let currentRange = selection.rangeCount == 1 ? selection.getRangeAt(0) : null;
+                    if (
+                        (skipSameRange || Browser.isEdge) &&
+                        currentRange &&
+                        currentRange.startContainer == range.startContainer &&
+                        currentRange.startOffset == range.startOffset &&
+                        currentRange.endContainer == range.endContainer &&
+                        currentRange.endOffset == range.endOffset
+                    ) {
+                        needAddRange = false;
+                    } else {
+                        selection.removeAllRanges();
+                    }
+                } catch (e) {}
+            }
+
+            if (needAddRange) {
+                selection.addRange(range);
+            }
+
+            if (!hasFocus(core)) {
+                core.cachedSelectionRange = range;
+            }
+
+            return true;
+        }
+    }
+
+    return false;
+};
+
+export default selectRange;
+
+/**
+ * @deprecated Only for compatibility with existing code, don't use ths function, use selectRange instead
+ */
+export const select: Select = (core: EditorCore, arg1: any, arg2?: any, arg3?: any, arg4?: any) => {
+    let range = arg1 instanceof Range ? arg1 : createRange(arg1, arg2, arg3, arg4);
+    return core.api.selectRange(core, range);
```

Comment:
Why do you no longer need the logic for when (arg1 instanceof Node)?

Reply:
It is checking arg1 is Range, not Node.
The old code support passing range as parameter, in that case we already have the range and no need to createRange

Reply:
I meant that in the old code, there was another conditional that checked if arg1 was a Node. But since this is now selectRange instead of select, it probably no longer applies.